main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java

```diff
@@ -47,33 +57,117 @@ public void validate(NoticeContainer noticeContainer) {
       for (int i = 1; i < shapeList.size(); ++i) {
         GtfsShape prev = shapeList.get(i - 1);
         GtfsShape curr = shapeList.get(i);
-        if (prev.hasShapeDistTraveled()
-            && curr.hasShapeDistTraveled()
-            && prev.shapeDistTraveled() >= curr.shapeDistTraveled()) {
+        if (!(prev.hasShapeDistTraveled() || curr.hasShapeDistTraveled())) {
+          continue;
+        }
+        if (prev.shapeDistTraveled() > curr.shapeDistTraveled()) {
+          noticeContainer.addValidationNotice(new DecreasingShapeDistanceNotice(prev, curr));
+          continue;
+        }
+        if (prev.shapeDistTraveled() != curr.shapeDistTraveled()) {
+          continue;
+        }
+        // equal distance and different coordinates
+        if (!areCloseWithDifferentCoordinates(curr, prev)) {
```

Comment:
Won't this branch execute when coordinates are identical OR where coordinates are > 1.11 m?

Which means the following `else if` will never execute because when `curr.shapePtLat() == prev.shapePtLat() && curr.shapePtLon() == prev.shapePtLon()` then `!areCloseWithDifferenteCoordinates` is true, so the previous branch has already executed.